### ISSUE TITLE
RUCSS - allow to safelist @font-face rules

### ISSUE THREAD

**Issue by vmanthos** [wp-media/wp-rocket#3963]
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version ✅
- Used the search feature to ensure that the bug hasn’t been reported before ✅

**Describe the bug**

Currently, the RUCSS safelist will accept CSS ids/classes/files. 

It's impossible to safelist `@font-face` rules like the following:
```CSS
@font-face{font-family:eicons;src:url(https://example.com/wp-content/plugins/elementor/assets/lib/eicons/fonts/eicons.eot?5.10.0);src:url(https://example.com/wp-content/plugins/elementor/assets/lib/eicons/fonts/eicons.eot?5.10.0#iefix) format("embedded-opentype"),url(https://example.com/wp-content/plugins/elementor/assets/lib/eicons/fonts/eicons.woff2?5.10.0) format("woff2"),url(https://example.com/wp-content/plugins/elementor/assets/lib/eicons/fonts/eicons.woff?5.10.0) format("woff"),url(https://example.com/wp-content/plugins/elementor/assets/lib/eicons/fonts/eicons.ttf?5.10.0) format("truetype"),url(https://example.com/wp-content/plugins/elementor/assets/lib/eicons/fonts/eicons.svg?5.10.0#eicon) format("svg");font-weight:400;font-style:normal}
```

This may result in fonts not being loaded on the front-end if that part of CSS is not added in the used CSS by the SaaS.

**To Reproduce**

Not relevant in this case.

**Expected behavior**

There should be a way to safelist `@font-face` rules.

**Screenshots**

**Additional context**

**Slack convo:** https://wp-media.slack.com/archives/C08N8J6VC/p1622185641187300?thread_ts=1622152050.184400&cid=C08N8J6VC

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

**Comment by gayanchandrab** [comment]
This is exactly what happens when Better Font Awesome plugin is used with RUCSS.

**Comment by iCaspar** [comment]
This issue will move to the treeshaker repo.

@GeekPress @vmanthos 
I'm picturing that implementation in the safelist UI field and corresponding treeshaker compatibilities, this would look like
```
@font_face{font-family:eicons}
```
with one font-family per line.

Grooming:
**Reproduced**
**Root Cause**
Feature is not implemented yet.

**Scope for Solution**
Step 0 - Add a testcase to fixture for `Shaker::shakeFontFaces()` method to verify.
Step 1 - Add array for `fontFamilies` to Treeshaker's `Safelist` class.
Step 2 - Add handling for parsing `@font_face{font-family:xxxxx}` as a font-family to add to the new array to the `addUnparsedSafelistItems()` method, and for adding directly to the array in `addParsedSafelistItems()` method.
Step 3 - Add `isSafeFontFamily(string $fontFamily)` method to the `Safelist` class to check safelist status for font families. (See `isSafeSelector()` for model.)
Step 4 - In the `Shaker` class, add a call to `Safelist::isSafeFontFamily()` just before https://github.com/wp-media/tree-shaker/blob/8f4ca259d403c63e59374e90aa19381c3fe87413/app/Services/Shaker.php#L325 where a true return adds the family to `$this->used_font_family` and continue.

**Estimated Effort**
[M] --  A few steps here, and finding the regex to do the parsing, but nothing in itself very hard.

**Comment by vmanthos** [comment]
>I'm picturing that implementation in the safelist UI field and corresponding treeshaker compatibilities, will with one font-family per line.

This looks good to me. 👍

**Comment by vmanthos** [comment]
Related ticket: https://secure.helpscout.net/conversation/1650940335/299200/

**Comment by DahmaniAdame** [comment]
https://secure.helpscout.net/conversation/1656517499/300403

**Comment by DahmaniAdame** [comment]
Related - https://secure.helpscout.net/conversation/1749608997/317830

**Comment by piotrbak** [comment]
Fixed with 3.11:
https://github.com/wp-media/nodejs-treeshaker/pull/55
